yt_dlp/extractor/veo.py

```diff
@@ -39,39 +42,33 @@ def _real_extract(self, url):
         video_data = self._download_json(
             'https://app.veo.co/api/app/matches/%s/videos' % video_id, video_id, 'Downloading video data')
 
-        title = metadata.get('title')
-        thumbnail = url_or_none(metadata.get('thumbnail'))
-
-        timestamp = unified_timestamp(metadata.get('created'))
-        duration = int_or_none(metadata.get('duration'))
-        view_count = int_or_none(metadata.get('view_count'))
-
         formats = []
         for fmt in video_data:
-            mimetype = fmt.get('mime_type')
+            mimetype = str_or_none(fmt.get('mime_type'))
+            format_url = url_or_none(fmt.get('url'))
             # skip configuration file for panoramic video
-            if mimetype == 'video/mp2t':
+            if not format_url or mimetype == 'video/mp2t':
                 continue
             height = int_or_none(fmt.get('height'))
-            bitrate = int_or_none(fmt.get('bit_rate'), scale=1000)
-            render_type = fmt.get('render_type')
+            format_id = join_nonempty(fmt.get('render_type'), height)
             formats.append({
-                'url': url_or_none(fmt.get('url')),
-                'format_id': '%s-%sp' % (render_type, height),
+                'url': format_url,
+                'format_id': format_id + 'p' if format_id else None,
                 'ext': mimetype2ext(mimetype),
                 'width': int_or_none(fmt.get('width')),
                 'height': height,
-                'vbr': bitrate
+                'vbr': int_or_none(fmt.get('bit_rate'), scale=1000),
             })
 
+        self._check_formats(formats, video_id)
```

Comment:
Is this really needed? It is generally preferable to use `--check-formats`

Reply:
Without it, downloads will fail for users not using `--check-formats`
```
$ yt-dlp https://app.veo.co/matches/20201027-last-period/
[Veo] 20201027-last-period: Downloading JSON metadata
[Veo] 20201027-last-period: Downloading video data
[info] 20201027-last-period: Downloading 1 format(s): panorama-2048p
[download] Resuming download at byte 261120
ERROR: unable to download video data: HTTP Error 403: Forbidden
```

Reply:
Personally, I think it is best if the option works properly even if it is not added.